### ISSUE TITLE
cannot build if using clang

### ISSUE THREAD

**Issue by braindevices** [cython/cython#2472]
It will raise any error: `unsupported option '--lto-partition=none'`

**Comment by gabrieldemarmiesse** [comment]
Cython is automatically tested against Clang with travis.ci and those tests are passing for pipy releases. Which means that the bug you mention is not trivial to reproduce. Could you provide as much details as possible to help us reproduce the bug?

**Comment by braindevices** [comment]
After digging deeper, I found this only happened in some system for example fedora. In those system, the Makefile from python config contain some flag like this:
`LDFLAGS="$(LDFLAGS) $(PGO_PROF_GEN_FLAG) -flto -fuse-linker-plugin -ffat-lto-objects -flto-partition=none"`
Thus resulting the error, if we manually set CC to clang. 
So I am sorry. This is not related to cython. More like setputools/distutils. The default setputools/distutils behaviour is using default LDSHARED even when `export CC=clang`. But if give `CXX=clang++` in env and `language="c++"` in extension, the linker switch to clang++ but still using all the flags. I guess unset any CC/CXX variables in env is the best choice.
